### PR TITLE
fix: APP-2338 - Fix dialog handling on deposit dialog

### PR DESCRIPTION
**Pull request by cgero-eth**
## Description

- Go back to previous page when opening the new-withdrawal page from the deposit dialog being connected to the wrong network and closing the network dialog
- Correctly display the network dialog when clicking on "connect" from the deposit dialog being connected to the wrong network

Task: [APP-2338](https://aragonassociation.atlassian.net/browse/APP-2338)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [x] I have tested my code on the test network.

[APP-2338]: https://aragonassociation.atlassian.net/browse/APP-2338?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/containers/transactionModals/depositModal.tsx [hunk 1]
```diff
@@ -75,14 +40,21 @@ const DepositModal: React.FC = () => {
...
   const handleConnectClick = useCallback(() => {
-    close();
-    loginFlowTriggeredRef.current = true;
-  }, [close]);
+    if (!isConnected) {
+      const modalState = {onSuccess: () => open('deposit')};
+      open('wallet', modalState);
+    } else if (isOnWrongNetwork) {
+      const modalState = {onSuccess: () => open('deposit')};
+      open('network', modalState);
```
Comment (Fabricevladimir) [inline src/containers/transactionModals/depositModal.tsx]: Slight improvement maybe. Considering the deposit modal is being opened on success, regardless, you can define `modalState` outside of the if block.

File: src/containers/walletMenu/LoginRequired.tsx [hunk 2]
```diff
@@ -16,36 +16,40 @@ import {useGlobalModalContext} from 'context/globalModals';
...
+    <ModalBottomSheetSwitcher isOpen={isOpen} onClose={onClose}>
       <ModalHeader>
         <Title>{t('alert.loginRequired.headerTitle')}</Title>
         {isDesktop && (
           <ButtonIcon
             mode="ghost"
             icon={<IconClose />}
             size="small"
-            onClick={handleClose}
+            onClick={() => onClose()}
```
Comment (Fabricevladimir) [inline src/containers/walletMenu/LoginRequired.tsx:52]: You can just pass this the function like so I think:
```jsx
onClick={onClose}
```

**Review by Fabricevladimir — COMMENTED** [review]
Sweet sweet refactor!

File: src/containers/transactionModals/depositModal.tsx (hunk 1 above)
Comment (cgero-eth) [reply src/containers/transactionModals/depositModal.tsx]: Makes sense 👍🏻

File: src/containers/walletMenu/LoginRequired.tsx (hunk 2 above)
Comment (cgero-eth) [reply src/containers/walletMenu/LoginRequired.tsx:52]: Unfortunately the `onClose` function expects an optional function as parameter (callback called when the modal is closed) while the `onClick` prop expects the on-click event, if we try to use `onClick={onClose}` we'll get a type error

Comment (Fabricevladimir) [reply src/containers/walletMenu/LoginRequired.tsx:52]: I see. Thanks

**Review by Fabricevladimir — APPROVED** [review]

**Pull request merged**